File: Python/program/ParkingLot.py
from Car import Car
from Motorcycle import Motorcycle

class ParkingLot:
    __kapasitas = 0
    __jmlKendaraanNow = 0
    __daftarKendaraan = list()

    def __init__(self, kapasitas=0, jmlKendaraanNow=0, daftarKendaraan=list()):
        self.__kapasitas = kapasitas
        self.__jmlKendaraanNow = jmlKendaraanNow
        self.__daftarKendaraan = daftarKendaraan
        if(len(self.__daftarKendaraan) != 0 and len(self.__daftarKendaraan) > self.__jmlKendaraanNow):
            self.__jmlKendaraanNow = len(self.__daftarKendaraan)

    
    def setKapasitas(self, kapasitas):
        self.__kapasitas = kapasitas
    
    def setJmlKendaraanNow(self, jmlKendaraanNow):
        self.__jmlKendaraanNow = jmlKendaraanNow

    def setDaftarKendaraan(self, daftarKendaraan):
        self.__daftarKendaraan = daftarKendaraan
    
    def getKapasitas(self):
        return self.__kapasitas
    
    def getJmlKendaraanNow(self):
        return self.__jmlKendaraanNow
    
    def getDaftarKendaraan(self):
        return self.__daftarKendaraan
    
    def getByColName(self, colName=""):
        if colName == "kapasitas":
            return self.getKapasitas()
        elif colName == "jmlKendaraanNow":
            return self.getJmlKendaraanNow()
        elif colName == "daftarKendaraan":
            return self.getDaftarKendaraan()
    
    def addKendaraan(self, vehicle):
        exist = False
        for i in self.__daftarKendaraan:
            if i.getPlatNomor() == vehicle.getPlatNomor():
                exist = True
        if(not exist):
           self.__daftarKendaraan.append(vehicle)
           self.__jmlKendaraanNow += 1
        else:
            print("Kendaraan dengan plat nomor " + vehicle.getPlatNomor() + " sudah ada!")

    def removeKendaraan(self, platNomor):
        for i in self.__daftarKendaraan:
            if i.__getPlatNomor() == platNomor:
                self.__daftarKendaraan.remove(i)
                self.__jmlKendaraanNow -= 1

File: CPP/ParkingLot.cpp
#pragma once
#include <iostream>
#include <string>
#include <bits/stdc++.h>
#include "Vehicle.cpp"

using namespace std;

class ParkingLot{
    private:
        int kapasitas, jmlKendaraanNow;
        vector<Vehicle*> daftarKendaraan;
    public:
        ParkingLot(){

        }
        ParkingLot(int kapasitas){
            this->kapasitas = kapasitas;
            this->jmlKendaraanNow;
            this->daftarKendaraan;
        }
        ParkingLot(int kapasitas, int jmlKendaraanNow){
            this->kapasitas = kapasitas;
            this->jmlKendaraanNow = jmlKendaraanNow;
            this->daftarKendaraan;
        }
        ParkingLot(int kapasitas, int jmlKendaraanNow, vector<Vehicle*> daftarKendaraan){
            this->kapasitas = kapasitas;
            this->jmlKendaraanNow = jmlKendaraanNow;
            this->daftarKendaraan = daftarKendaraan;
        }

        void setKapasitas(int kapasitas){
            this->kapasitas = kapasitas;
        }
        void setJmlKendaraanNow(int jmlKendaraanNow){
            this->jmlKendaraanNow = jmlKendaraanNow;
        }
        void setDaftarKendaraan(vector<Vehicle*> daftarKendaraan){
            this->daftarKendaraan = daftarKendaraan;
        }

        int getKapasitas(){
            return this->kapasitas;
        }
        int getJmlKendaraanNow(){
            return this->jmlKendaraanNow;
        }
        vector<Vehicle*> getDaftarKendaraan(){
            return this->daftarKendaraan;
        }

        string getByColName(string colName){
            if(colName == "kapasitas"){
                return to_string(this->getKapasitas());
            }
            else if(colName == "jmlKendaraanNow"){
                return to_string(this->getJmlKendaraanNow());
            }
            else if(colName == "daftarKendaraan"){
                return "list";
            }
        }
        vector<Vehicle*> getByColNameList(string colName){
            if(colName == "daftarKendaraan"){
                return this->getDaftarKendaraan();
            }
        }

        void addKendaraan(Vehicle v1){
            bool exist = false;
            for(vector<Vehicle*>::iterator it = this->daftarKendaraan.begin();it != this->daftarKendaraan.end();it++){
                if(v1.getPlatNomor() == (*it)->getPlatNomor()){
                    exist = true;
                }
            }
            if(!exist){
                this->daftarKendaraan.push_back(&v1);
            }
            else{
                cout << "Kendaraan dengan plat nomor : " << v1.getPlatNomor() << " sudah ada\n";
            }
        }

        void removeKendaraan(string platNomor){
            vector<Vehicle*>::iterator deleted = this->daftarKendaraan.begin();
            bool exist = false;
            for(vector<Vehicle*>::iterator it = this->daftarKendaraan.begin();it != this->daftarKendaraan.end();it++){
                if(platNomor == (*it)->getPlatNomor()){
                    deleted = it;
                    exist = true;
                }
            }
            if(!exist){
                cout << "Kendaraan dengan plat nomor : " << platNomor << " tidak ada!\n";
            }
            else{
                this->daftarKendaraan.erase(deleted);
            }
        }
};